### DISCUSSION TITLE
Docker container introspect url returns active:false

### DISCUSSION THREAD

**Discussion by ale0hub** [keycloak/keycloak#21795] [Q&A]
Hello, 

I have a docker container running Keycloak with a mapped external port on my computer. 

I manage to request tokens to the external port from a public client (in local network) using authorization code flow.

Afterwards, I introspect a token to the external port and I get its details and whether it is active or not.

Unfortunately, if I introspect it to the internal port of the docker container I always get {active:false} even if the token has expired or not.

I am suspecting that the token must be issued from the same host (or client) where I am inspecting it but as I understand that should not matter since authorization is successfull (I receive 200 OK) and I am in "talking" to the same realm.

The introspect call to the external port or to the internal port is done in the same way, by basic Authorization for a confidential client. 

I want to clarify that for both calls I am using the same realm, but different client and different host.

How can I be sure of what is missing? And what can I do to overcome that?

Thanks everyone in advance

**Comment by Neha-Saxena15** [comment]
@ale0hub did you find the solution for this ?

**Comment by netstuff** [comment]
@ale0hub @Neha-Saxena15 solution [is here](https://stackoverflow.com/a/62310113/26360864)

**Short summary**: in my case I have got access token on host-machine's `localhost:8080`, but introspect the same token on `keycloak:8080`. I get solution with forced setting header `Host` to original `localhost:8080` value
